### PR TITLE
feat: Update TF and modules

### PR DESCRIPTION
**Pull request by ryanfaircloth**
This PR makes minimal changes to update the TF code to use the current version of terraform and modules/providers.

This PR Also ensures the caller user creating the stack AND the root account of the aws org has access to the cluster

### CODE REVIEW

File: aws/eks.tf [hunk 1]
```diff
@@ -14,8 +10,7 @@ provider "kubernetes" {
   host                   = data.aws_eks_cluster.cluster.endpoint
   cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
   token                  = data.aws_eks_cluster_auth.cluster.token
-  load_config_file       = false
-  version                = "~> 1.11"
+  #load_config_file       = false
```
Comment (jswoods) [inline aws/eks.tf]:
```suggestion
```

File: aws/eks.tf [hunk 2]
```diff
@@ -37,32 +32,78 @@ resource "aws_security_group" "eks_msk" {
...
+    },
   ]
+  aws_auth_accounts = [
+    data.aws_caller_identity.current.account_id
+  ]
+
+  # tags = {
+  #   Name = local.cluster_name
+  # }
+
```
Comment (jswoods) [inline aws/eks.tf]:
```suggestion
```

File: aws/main.tf [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
 provider "aws" {
-  version = ">= 2.28.1"
-  region  = var.region
+  # ... other configuration ...
+  default_tags {
+    tags = {
+      Name        = local.cluster_name
+      Environment = var.environment
+      Owner       = var.owner
+      App         = "aws2humio"
```
Comment (jswoods) [inline aws/main.tf]: What is aws2humio?

File: aws/main.tf [hunk 4]
```diff
@@ -1,19 +1,38 @@
...
+      #repo          = "https://github.com/humio-contrib/humio-aws-cloutrail-tf"
```
Comment (jswoods) [inline aws/main.tf]:
```suggestion
```

File: aws/variables.tf [hunk 5]
```diff
@@ -12,15 +12,15 @@ variable "humio_instance_count" {
 }
 variable "kafka_instance_type" {
   type    = string
-  default = "kafka.m5.large"
```
Comment (jswoods) [inline aws/variables.tf]: I think we want to keep this as m5.large. The t3s run out of credits very fast if there is any kind of load thrown at them.

File: aws/vpc.tf [hunk 6]
```diff
@@ -3,9 +3,9 @@ data "aws_availability_zones" "available" {
 
 module "vpc" {
   source  = "terraform-aws-modules/vpc/aws"
-  version = "2.47.0"
+  version = "3.14.0"
 
-  name                 = local.cluster_name
+  #name                 = local.cluster_name
```
Comment (jswoods) [inline aws/vpc.tf:8]: Any reason we don't want to name the vpc?

Comment (ryanfaircloth) [reply aws/vpc.tf:8]: It conflicts with the provider default tags and throws and error when the name is the same

File: aws/main.tf (hunk 3 above)
Comment (ryanfaircloth) [reply aws/main.tf]: cruft fixed

File: aws/variables.tf (hunk 5 above)
Comment (ryanfaircloth) [reply aws/variables.tf]: I should have changed that in a local variables so I put it back. It worth pointing out in docs that default sizes are probably not right for anyone instance sizes and counts need to be calculated based on needs

**Review by jswoods — APPROVED** [review]

**Pull request merged**